### PR TITLE
Bump @material components to 0.27

### PR DESCRIPTION
**Pull request by steverep**
## Proposed change

Meant to replace #14982 from dependabot since they all will have the same minimal release notes. 👎🏻 

The commit history is more helpful, but a lot of it is just bumping the MDC sub-dependencies as v14 progressed.  Occasionally, there are commits not mentioned in the release notes (e.g. Bram's fix for the icon button, which is why I removed the patch).  I'll do some more functional testing tomorrow, but eyes on any breaking style changes would help please.

I also fixed an error I made in #14567 for `.gitignore`.

## Type of change

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by steverep** [comment]
I did more functional testing and went through all the non-MDC bump commits.  I'm not finding any issues. 👍🏻

File: package.json [hunk 1]
```diff
@@ -47,30 +47,30 @@
...
-    "@material/mwc-slider": "0.25.3",
-    "@material/mwc-switch": "0.25.3",
-    "@material/mwc-tab": "0.25.3",
-    "@material/mwc-tab-bar": "0.25.3",
-    "@material/mwc-textarea": "^0.25.3",
-    "@material/mwc-textfield": "0.25.3",
-    "@material/mwc-top-app-bar-fixed": "^0.25.3",
-    "@material/top-app-bar": "14.0.0-canary.261f2db59.0",
+    "@material/chips": "^14.0.0",
+    "@material/data-table": "^14.0.0",
```
Comment (bramkragten) [inline package.json:51]: these need to be pinned to the same version as used in `mwc`

File: package.json [hunk 2]
```diff
@@ -47,30 +47,30 @@
...
+    "@material/mwc-ripple": "^0.27.0",
+    "@material/mwc-select": "^0.27.0",
+    "@material/mwc-slider": "^0.27.0",
+    "@material/mwc-switch": "^0.27.0",
+    "@material/mwc-tab": "^0.27.0",
+    "@material/mwc-tab-bar": "^0.27.0",
+    "@material/mwc-textarea": "^0.27.0",
+    "@material/mwc-textfield": "^0.27.0",
+    "@material/mwc-top-app-bar-fixed": "^0.27.0",
+    "@material/top-app-bar": "^14.0.0",
```
Comment (bramkragten) [inline package.json:73]: this too

File: yarn.lock [hunk 3]
```diff
@@ -1966,136 +1984,166 @@ __metadata:
...
   languageName: node
   linkType: hard
 
-"@material/checkbox@npm:14.0.0-canary.261f2db59.0":
-  version: 14.0.0-canary.261f2db59.0
-  resolution: "@material/checkbox@npm:14.0.0-canary.261f2db59.0"
+"@material/base@npm:^14.0.0":
+  version: 14.0.0
+  resolution: "@material/base@npm:14.0.0"
   dependencies:
```
Comment (bramkragten) [inline yarn.lock:1999]: base is included twice now for example.

File: package.json (hunk 1 above)
Comment (steverep) [reply package.json]: Fixed

**Review by bramkragten — APPROVED** [review]

**Pull request merged**